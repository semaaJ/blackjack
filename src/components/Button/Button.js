import { ColorSwatchIcon, RefreshIcon, EyeIcon, PlayIcon } from '@heroicons/react/solid';

export const ICONS = {
    0: <ColorSwatchIcon className="ml-2 h-5 w-5 text-gray-500" />,
    1: <PlayIcon className="ml-2 h-5 w-5 text-gray-500" />,
    2: <h1 className="ml-2 text-base text-gray-500">x2</h1>,              
    3: <EyeIcon className="ml-2 h-5 w-5 text-gray-500" />,            
    4: <RefreshIcon className="ml-2 h-5 w-5 text-gray-500" />                            
}

const Button = (props) => {
    const { id, icon, text, disabled, onClick } = props;
    return (
        <button 
            id={id}
            disabled={disabled}
            onClick={onClick} 
            className="w-btn-small h-btn text-center text-gray-700 text-sm bg-white border-white 
                sm:w-full sm:font-semibold sm:text-lg hover:bg-gray-200 disabled:bg-gray-200">
               <div className="flex items-center justify-center">
                   { text }
                   { ICONS[icon]}             
                </div>
        </button>
    )
}

export default Button;