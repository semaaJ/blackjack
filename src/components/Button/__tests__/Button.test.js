import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';


it('should render', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with text', () => {
    const tree = renderer.create(<Button text="name" />).toJSON();
    expect(tree).toMatchSnapshot();
});

  
it('should render disabled when disabled prop passed', () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});


