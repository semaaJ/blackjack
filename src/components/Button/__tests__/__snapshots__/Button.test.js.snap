// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<button
  className="w-btn-small h-btn text-center text-gray-700 text-sm bg-white border-white 
 sm:w-full sm:font-semibold sm:text-lg hover:bg-gray-200 disabled:bg-gray-200"
>
  <div
    className="flex items-center justify-center"
  />
</button>
`;

exports[`should render disabled when disabled prop passed 1`] = `
<button
  className="w-btn-small h-btn text-center text-gray-700 text-sm bg-white border-white 
 sm:w-full sm:font-semibold sm:text-lg hover:bg-gray-200 disabled:bg-gray-200"
  disabled={true}
>
  <div
    className="flex items-center justify-center"
  />
</button>
`;

exports[`should render with text 1`] = `
<button
  className="w-btn-small h-btn text-center text-gray-700 text-sm bg-white border-white 
 sm:w-full sm:font-semibold sm:text-lg hover:bg-gray-200 disabled:bg-gray-200"
>
  <div
    className="flex items-center justify-center"
  >
    name
  </div>
</button>
`;
